light-client/src/macros.rs

```diff
@@ -0,0 +1,17 @@
+/// Bail out of the current function with the given error kind.
+#[macro_export]
+macro_rules! bail {
+    ($kind:expr) => {
+        return Err($kind.into());
+    };
+}
+
+/// Ensure a condition holds, returning an error if it doesn't (ala `assert`).
+#[macro_export]
+macro_rules! ensure {
+    ($cond:expr, $kind:expr) => {
+        if !($cond) {
+            return Err($kind.into());
+        }
+    };
+}
```

Comment:
Just noting, anomaly, which seems to be already used throughout this module, provides almost the same macros:
https://github.com/iqlusioninc/crates/blob/develop/anomaly/src/macros.rs#L34-L44

Maybe there are good reasons to define your own here. Not sure.

Reply:
Yeah I saw those, but since we are using `thiserror` we do not need to specify an additional message for the error, as it will pick up the one defined in the `ErrorKind` Display implementation via the `#[error]` attribute. But if you see valuable information we could append to the error message via this extra argument in the original `anomaly` macros, let me know and I'd gladly update the code and get rid of the custom macros.